### DISCUSSION TITLE
Default Eloquent Cast

### DISCUSSION THREAD

**Discussion by aaronflorey** [spatie/laravel-data#265] [Ideas]
Currently when using a DTO as a cast in Eloquent, if the column is empty, it won't cast the object.

I'm wanting it to default to an empty DTO if the column is empty, would a PR be accepted to add this functionality possibly through an Attribute? `#[EloquentCastDefault]` or something

**Comment by Nir-An** [comment]
Hi @aaronflorey :+1: 

if I understand your question/idea

I think the ``source of true`` should come from the model (and only from the model)
 and not from the object that is based on the model  

as a result, the default value should be created only by the model


```php

namespace App\Models;
 
use Illuminate\Database\Eloquent\Model;
 
class Flight extends Model
{
    /**
     * The model's default values for attributes.
     *
     * @var array
     */
    protected $attributes = [
        'delayed' => false,
    ];
}

  /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'delayed' => 'boolean',
    ];

```

**Comment by aaronflorey** [comment]
Hey @Nir-An,

Thanks for the reply! 

Yeah, that's one way to do it, but unfortunately it's not a great solution for this package since you can't use expressions in array properties. I can set the attributes to an empty array to achieve the same effect but not ideal.

Thanks

**Comment by rubenvanassche** [comment]
@aaronflorey 

Any ideas how you would implement such functionality? At the moment not completely against to it but I don't know how you would pull it off.

**Reply by aaronflorey** [reply]
I would probably have an attribute be applied on the DTO, and then check for the attribute's existence in DataEloquentCast, but I'm just thinking now that we may not want to do reflection on every attribute cast.

I found an okay solution of defaulting the column as '{}' in the database, or putting `'dto' => '{}'` in `$attributes` on the model.

**Comment by Nir-An** [comment]
> I found an okay solution of defaulting the column as '{}' in the database or putting 'dto' => '{}' in $attributes on the model.

in the Database, to make your life easier I encourage you to keep an empty ``column`` as ``null`` by default, and not ``{}`` by default. 

 putting 'dto' => '{}' in $attributes on the model should be better solution.
